File: src/avaliacaoPPGI/PublicacaoPeriodico.java
package avaliacaoPPGI;

import java.util.ArrayList;

class PublicacaoPeriodico extends Publicacao {

	private int numero;
	private int volume;
	
	public PublicacaoPeriodico(int ano, Veiculo veiculo, String titulo, ArrayList<Docente> autores, 
			int numero, int volume, int paginaInicial, int paginaFinal) {
		super(ano, veiculo, titulo, autores, paginaInicial, paginaFinal);
		this.numero = numero;
		this.volume = volume;
	}
	
	public int getNumero() {
		return numero;
	}
	public void setNumero(int numero) {
		this.numero = numero;
	}
	
	public int getVolume() {
		return volume;
	}
	
	public void setVolume(int volume) {
		this.volume = volume;
	}

	@Override
	public String toString() {
		return "PublicacaoPeriodico [numero=" + numero + ", volume=" + volume + ", ano=" + ano + ", veiculo=" + veiculo
				+ ", titulo=" + titulo + ", autores=" + autores + ", paginaInicial=" + paginaInicial + ", paginaFinal="
				+ paginaFinal + "]";
	}

}

File: julia/cod/PublicacaoPeriodico.jl
using Base
include("Publicacao.jl")

struct PublicacaoPeriodico <: Publicacao
    @atributos_Publicacao
    numero::Int64
    volume::Int64

    PublicacaoPeriodico(ano::Int, veiculo::Veiculo, titulo::String,
    autores::Array{Docente}, pg_inicial::Int64, pg_final::Int64,
    numero::Int64, volume::Int64) = new(Date(ano), veiculo, titulo,
    autores, pg_inicial, pg_final, numero, volume)
end

numero(p::PublicacaoPeriodico) = p.numero
volume(p::PublicacaoPeriodico) = p.volume